From f49162600c5ed54341ebeba9bc773fefc6b8153d Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Tue, 7 Nov 2017 10:07:03 +0000
Subject: [PATCH] v2v: vddk: Update minimum version of nbdkit to 1.1.16.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

While 1.1.14 will technically still work, 1.1.16 adds the
‘selinux=yes|no’ flag to ‘--dump-config’ output.  It's also the more
widely available and tested version at the time of writing.

Thanks: Ming Xie.
(cherry picked from commit 415e9d82e901647438f68e4748720d88153778aa)
---
 v2v/input_libvirt_vddk.ml | 5 +++--
 v2v/virt-v2v.pod          | 2 +-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/v2v/input_libvirt_vddk.ml b/v2v/input_libvirt_vddk.ml
index 33b34bd2e..d1f2ae360 100644
--- a/v2v/input_libvirt_vddk.ml
+++ b/v2v/input_libvirt_vddk.ml
@@ -61,13 +61,14 @@ class input_libvirt_vddk vddk_options password libvirt_uri parsed_uri guest =
 
     (* Check it's a new enough version.  The latest features we
      * require are ‘--exit-with-parent’ and ‘--selinux-label’, both
-     * added in 1.1.14.
+     * added in 1.1.14.  (We use 1.1.16 as the minimum here because
+     * it also adds the selinux=yes|no flag in --dump-config).
      *)
     let lines = external_command "nbdkit --help" in
     let lines = String.concat " " lines in
     if String.find lines "exit-with-parent" == -1 ||
        String.find lines "selinux-label" == -1 then
-      error (f_"nbdkit is not new enough, you need to upgrade to nbdkit ≥ 1.1.14")
+      error (f_"nbdkit is not new enough, you need to upgrade to nbdkit ≥ 1.1.16")
   in
 
   (* Check that the VDDK plugin is installed and working *)
diff --git a/v2v/virt-v2v.pod b/v2v/virt-v2v.pod
index 4f7ea8da6..48a20cccc 100644
--- a/v2v/virt-v2v.pod
+++ b/v2v/virt-v2v.pod
@@ -1305,7 +1305,7 @@ library is permitted by the license.
 =item 2.
 
 You must also compile nbdkit, enabling the VDDK plugin.  At least
-nbdkit E<ge> 1.1.14 is required, but it is usually best to compile
+nbdkit E<ge> 1.1.16 is required, but it is usually best to compile
 from the git tree.
 
 =over 4
-- 
2.14.3

